### PR TITLE
Created an initial implementation of evaluation in dmn engine. Resolv…

### PR DESCRIPTION
**Pull request by mariuszkumor**
…ed #40

### CODE REVIEW

**Comment by mariuszkumor** [comment]
Without unit tests in this pull request because the final internal API will be clarified during implementation of  more complex evaluation provider like GroovyExpressionEvaluationProvider.

**Comment by stlhrt** [comment]
Doesn't build...

**Comment by mariuszkumor** [comment]
I test only the most important stuff for now, since I think lot of changes will be soon.

File: src/main/java/org/powerflows/dmn/engine/evaluator/entry/EntryEvaluator.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        return expressionType;
+    }
+
+    private ExpressionType resolveExpressionType(final OutputEntry outputEntry, final Decision decision) {
+        final ExpressionType expressionType;
+
+        if (outputEntry.getExpression() != null && outputEntry.getExpression().getType() != null) {
+            expressionType = outputEntry.getExpression().getType();
+        } else if (decision.getExpressionType() != null) {
+            expressionType = decision.getExpressionType();
```
[the remark this answers is not in the document]
Comment (mariuszkumor) [reply src/main/java/org/powerflows/dmn/engine/evaluator/entry/EntryEvaluator.java]: Form the other side, I agree, we should ot keep this logi here. I remove and create the appropriate issue for builders improvement.

File: src/main/java/org/powerflows/dmn/engine/evaluator/entry/expression/provider/LiteralExpressionEvaluationProvider.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        } else {
+            evaluationResult = false;
+        }
+
+        return evaluationResult;
+    }
+
+    @Override
+    public Object evaluateInput(final Input input, final ModifiableContextVariables contextVariables) {
+        throw new EvaluationException("Input evaluation is not possible for literal expression.");
```
Comment (stlhrt) [inline src/main/java/org/powerflows/dmn/engine/evaluator/entry/expression/provider/LiteralExpressionEvaluationProvider.java]: it is supported, its just taking of named variable from user provided context

**Pull request merged**